app/models/concerns/manage_user_roles.rb

```diff
@@ -36,20 +36,8 @@ def max_role_level
       self.roles.maximum("level") || 0
     end
 
-    def create_or_remove_user_roles(role_ids)
-      role_ids = role_ids || []
-      remove_old_user_roles(role_ids)
-
-      role_ids.each do |role_id|
-        user_roles.where(role_id: role_id).first_or_create
-      end
-    end
-
-    private
-
-    def remove_old_user_roles(role_ids)
-      role_ids_to_remove = roles.pluck(:id) - role_ids
-      user_roles.where("role_id IN(?)", role_ids_to_remove).destroy_all
+    def create_or_remove_user_roles(user, role_ids)
```

Comment:
great! given that the simplification works and it's just one line, why not merge it into `update_roles_for_user` and delete `create_or_remove_user_roles`